### ISSUE TITLE
Add more details around what it means to use Ether units

### ISSUE THREAD

**Issue by logankilpatrick** [ethereum/solidity#11101]
## Abstract

Right now, it is rather unclear based on reading the docs here: https://github.com/ethereum/solidity/blob/v0.8.2/docs/units-and-global-variables.rst what we can actually use ether units for. 

## Motivation

Adding in more examples of how it can be used. I was specifically interested in how to solve the division/floating point problem in Solidity and was pointed to this section but it is not clear how I can actually use those ether units. 

## Backwards Compatibility

After updating the docs for 8.2, it should be propagated to older versions.

**Comment by cameel** [comment]
Thanks for your input! We definitely want the docs to be clear so if there's something we can do to improve them, we'll work on that. That being said, I think that the example shows pretty clearly that numbers with these units are just integers:

> assert(1 wei == 1);
> assert(1 gwei == 1e9);
> assert(1 ether == 1e18);
> 
> The only effect of the subdenomination suffix is a multiplication by a power of ten.

I.e. `1 wei` is exactly the same as `1` and `wei` is there just for clarity. `1 gwei` is a shorthand for `1000000000`, etc. There's nothing more to it and it has nothing to do with fixed-point arithmetic (there are no floating-point numbers in Solidity). Even something like `0.000001 ether` is in fact just an integer (equivalent to `1000 gwei` or `1000000000000`).

Could you give some examples of operations where the result you got was counterintuitive? Did you for example assume that `1 ether` means something like `1.0` and lower denominations are fractions? Or maybe that these units are typed and not just equivalent to integers? I think that most of the confusion regarding using these units in calculations results from the assumptions people come with. We could expand on it a bit (and maybe rephrase it if the current phrasing makes things less clear than they could be) but we have to know first which situations are problematic.
